include/dll/dbn_impl.hpp

```diff
@@ -2073,11 +2073,9 @@ struct dbn final {
             if (cpp_unlikely(!full_batch)) {
                 auto soutput = slice(output, 0, n);
 
-                batch_loss  = (1.0 / (2.0 * s)) * sum((soutput - labels) >> (soutput - labels));
-                batch_error = (1.0 / s) * asum(labels - soutput);
+                std::tie(batch_loss, batch_error) = etl::ml::mse(soutput, labels, 1, 1);
```

Comment:
I also set the both scaling factors (alpha and beta) to 1 here, is it correct?

Reply:
You should use the (1.0 / s) and (1.0 / (2.0 * s)  as the scaling factors, once you have removed the division by n in etl-gpu-blas.